x-pack/plugins/infra/public/pages/metrics/hosts/components/tabs/logs/logs_tab_content.tsx

```diff
@@ -36,6 +40,13 @@ export const LogsTabContent = () => {
     return filterQuery.query || hostsFilterQueryParam;
   }, [filterQuery.query, hostNodes]);
 
+  useEffect(() => {
+    telemetry.reportHostsViewLogsQuerySubmitted({
+      host_filters: createHostsFilterQueryParam(hostNodes),
+      query: filterQuery.query,
+    });
+  }, [filterQuery.query, hostNodes, telemetry]);
```

Comment:
**suggestion**: This event will be triggered way more than necessary as it watches every single change of the query text. Also, this is not a side effect, but an action triggered by a specific action, we should trigger this event when the change handler for the input text is triggered, or even better, trigger it only when the results of the log change after a query search on its input field.

Reply:
> this gets fired when either the host nodes change or the query changes, in these 2 cases we are already retrieving new log entries.

This is the point, what we want to understand is if users like or not that search bar. 
In a scenario where the user opens the logs tab, and starts filtering the hosts' table using controls, filters etc, we'll be still tracking this event even if they don't interact with the logs search bar, which can be misleading for the analysis on this event.

> I can't see how will it be different or less frequent if we do it on the input text change handler rather than the query change.

This is more about how React side effects should be used concerning what are the user actions on a page.
What we want to do is to record the event when the user type something on that input, not when a state changes. These are 2 different behaviours, what we'd like to have in a simplified example is:
```js
const debouncedTrackEvent = debounce((query) => {
  telemetry.reportHostsViewLogsQuerySubmitted({
    host_filters: createHostsFilterQueryParam(hostNodes),
    query,
  });
}, 500)


const handleInputChange = (event) => {
  setInputValue(event.target.value)
  debouncedTrackEvent(event.target.value)
}
```

This will prevent adding a side effect and triggering the event tracking only when the user interacts with the target feature.

Reply:
++ to only submitting the final query. 

Sending an event for each change to the query text (without being submitted) can lead us to wrong assumptions (the users don't really query that) and could quickly hit [FS's rate limit](https://developer.fullstory.com/browser/capture-events/analytics-events/#limits).

> What we want to do is to record the event when the user type something on that input, not when a state changes

IMO, we don't need a custom event for that. FullStory already collects it:

<img width="783" alt="image" src="https://user-images.githubusercontent.com/5469006/229771539-ce5f2329-c816-464d-b147-c8b83e53d97a.png">

cc @elastic/platform-analytics @shahinakmal

Reply:
> IMO, we don't need a custom event for that. FullStory already collects it:

Agree, but the event is currently tracking another property that we need to compose manually. I'm not sure we need to keep that prop, if we can remove it we can surely switch this to an automatic event collected by FullStory